test/struct.js

```diff
@@ -516,3 +519,35 @@ test('enforces ordinal identifiers', function t(assert) {
     }
     assert.end();
 });
+
+test('allows undefined as default', function t(assert) {
+    var argStruct = new ThriftStruct({strict: false, defaultValueDefinition: defaultValueDefinition});
+    var tMock = {
+        structs: {},
+        resolve: function resolve() {
+            return new ThriftBoolean();
+        },
+        resolveValue: function resolveValue(valueDef) {
+            assert.equal(valueDef.type, 'Literal', 'Value definition is a literal');
+            assert.true(valueDef.hasOwnProperty('value'), 'Has defined value property');
+            assert.true(valueDef.value === undefined, 'Value is undefined')
+        }
+    };
+    argStruct.compile({
+        id: {name: 'NewDefault'},
+        fields: [
+            {
+                id: {value: 1},
+                name: 'name',
+                valueType: {
+                    type: 'BaseType',
+                    baseType: 'boolean'
+                },
+                required: false,
+                optional: false
+            }
+        ]
+    }, {allowOptionalArguments: true});
+    argStruct.link(tMock);
+    assert.end();
```

Comment:
Can you verify that there already is a test that asserts the current behavior remains unchanged?